Assets/MixedRealityToolkit/Inspectors/Setup/BuildDeployWindow.cs

```diff
@@ -0,0 +1,1436 @@
+            }
+
+            EditorGUILayout.Space();
+        }
+
+        private static void RenderOpenVisualStudioButton()
+        {
+            if (GUILayout.Button("Open in Visual Studio", GUILayout.Width(HALF_WIDTH)))
+            {
+                string slnFilename = Path.Combine(BuildDeployPreferences.BuildDirectory, $"{PlayerSettings.productName}.sln");
+
+                if (File.Exists(slnFilename))
+                {
+                    EditorApplication.delayCall += () => Process.Start(new FileInfo(slnFilename).FullName);
+                }
+                else if (EditorUtility.DisplayDialog(
+                    "Solution Not Found",
+                    "We couldn't find the Project's Solution. Would you like to Build the project now?",
+                    "Yes, Build", "No"))
+                {
+                    EditorApplication.delayCall += BuildUnityProject;
+                }
+            }
+        }
+
+        private static void RenderPlayerSettingsButton()
+        {
+            if (GUILayout.Button("Open Player Settings"))
+            {
+                Selection.activeObject = Unsupported.GetSerializedAssetInterfaceSingleton("PlayerSettings");
+            }
+        }
+
+        private static void RenderOpenBuildDirectoryButton()
+        {
+            using (new EditorGUI.DisabledGroupScope(!Directory.Exists(BuildDeployPreferences.AbsoluteBuildDirectory)))
+            {
+                if (GUILayout.Button("Open", EditorStyles.miniButtonRight, GUILayout.Width(100f)))
+                {
+                    EditorApplication.delayCall += () => Process.Start(BuildDeployPreferences.AbsoluteBuildDirectory);
+                }
+            }
+        }
+
+        #endregion
+
+        #region Utilities
+
+        private void RemoveConnection()
+        {
+            var connections = portalConnections?.Connections;
+            if (connections != null && connections.Count > 0)
+            {
+                portalConnections.Connections.RemoveAt(CurrentConnectionIndex);
+                CurrentConnectionIndex--;
+                UpdatePortalConnections();
+            }
+        }
+
+        private void AddConnection()
+        {
+            DeviceInfo currentConnection = CurrentConnection;
+            portalConnections.Connections.Add(new DeviceInfo(EMPTY_IP_ADDRESS, currentConnection.User, currentConnection.Password));
+            CurrentConnectionIndex = portalConnections.Connections.Count - 1;
+            UpdatePortalConnections();
+        }
+
+        private async void ConnectToDevice()
+        {
+            DeviceInfo currentConnection = CurrentConnection;
+            var machineName = await DevicePortal.GetMachineNameAsync(currentConnection);
+            currentConnection.MachineName = machineName?.ComputerName;
+            UpdatePortalConnections();
+            Debug.Log($"Successfully connected to device {machineName?.ComputerName} with IP {currentConnection.IP}");
```

Comment:
Is this automatically a successful connection? Is there any failure case?

Reply:
If I use a random IP, and test the connection, I see a confusing combination of messages:

![image](https://user-images.githubusercontent.com/3580640/70936644-92e3b080-1ff7-11ea-8ecd-5fd66a242d13.png)